tt_metal/hw/inc/wormhole/noc_nonblocking_api.h

```diff
@@ -64,24 +67,28 @@ inline __attribute__((always_inline)) bool ncrisc_noc_reads_flushed(uint32_t noc
   return (NOC_STATUS_READ_REG(noc, NIU_MST_RD_RESP_RECEIVED) == noc_reads_num_issued[noc]);
 }
 
-inline __attribute__((always_inline)) void ncrisc_noc_fast_write(uint32_t noc, uint32_t cmd_buf, uint32_t src_addr, uint64_t dest_addr, uint32_t len_bytes, uint32_t vc, bool mcast, bool linked, uint32_t num_dests) {
+inline __attribute__((always_inline)) void ncrisc_noc_fast_write(uint32_t noc, uint32_t cmd_buf, uint32_t src_addr, uint64_t dest_addr, uint32_t len_bytes, uint32_t vc, bool mcast, bool linked, uint32_t num_dests, bool posted = false) {
```

Comment:
when do we use posted vs non-posted writes?  wonder if we want to pay the cost in the api of the test if differentiation is rare

Reply:
Presently no call from dataflow_api uses this, but I plan to add new ones for the optimized implementation of the packetized path. 

I don't think there is any cost to adding the parameter in the base NOC API call, since the default doesn't break any previous usage, and it will be constant-propagated in any case. 

Posted writes have no ack from the destination, and can be used whenever we're not interested in the end-to-end barrier (though we still have a flush barrier to ensure data have left the local L1). There are separate counters in NOC for posted and non-posted ones, so we can still have barriers that act on the latter but ignore the former.